src/services/service.ts

```diff
@@ -174,6 +176,11 @@ export default class Service {
             request.setPartialFailure(options.partial_failure)
         }
 
+        // Make all requests immutable if prevent_mutations is enabled
+        if (this.prevent_mutations) {
+            request.setValidateOnly(true)
```

Comment:
`setValidateOnly` is not always available on the request object. We need to add a check to confirm that it's possible to run this request with validateOnly, similar to line 167.

When the request doesn't support `validateOnly` and `prevent_mutations === true`, we should probably just return from the function without actually calling the service. We should probably also console.log to avoid frustration with a message like `${serviceName}.${method} was not called because prevent_mutations is true.`